### PR TITLE
[core] Fix some DashboardLayout bugs and make some docs examples more consis…

### PR DESCRIPTION
**Pull request by apedroferreira**
Fixes the issues mentioned here https://github.com/mui/mui-toolpad/pull/3893#issuecomment-2269324520:

- Fix scrollbar color not always being set when used in iframes with different color schemes (such as home page)
- Fix some possible issues when the layout is used in iframes (it required adding some styles that in most cases would not be needed, though, with `100vw` to set an explicit width)
- Make some docs examples more consistent with other examples in the page they're in

https://deploy-preview-3905--mui-toolpad-docs.netlify.app/toolpad
https://deploy-preview-3905--mui-toolpad-docs.netlify.app/toolpad/core/react-dashboard-layout/

### CODE REVIEW

File: docs/data/toolpad/core/components/account/AccountWithDashboard.js [hunk 1]
```diff
@@ -1,15 +1,63 @@
...
+import PropTypes from 'prop-types';
 import Box from '@mui/material/Box';
-import { AppProvider, DashboardLayout } from '@toolpad/core';
+import Typography from '@mui/material/Typography';
+import { extendTheme } from '@mui/material/styles';
+import DashboardIcon from '@mui/icons-material/Dashboard';
+import { AppProvider } from '@toolpad/core/AppProvider';
+import { DashboardLayout } from '@toolpad/core/DashboardLayout';
+
+const NAVIGATION = [
```
Comment (apedroferreira) [inline docs/data/toolpad/core/components/account/AccountWithDashboard.js:10]: Yeah it's adding all this boilerplate here too but this way it matches the other examples...

File: packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx [hunk 2]
```diff
@@ -466,9 +466,15 @@ function DashboardLayout(props: DashboardLayoutProps) {
...
+      <Box
+        component="main"
+        sx={{
+          flexGrow: 1,
+          minWidth: { xs: isMobileNavigationOpen ? '100vw' : 'auto', md: 'auto' },
+        }}
+      >
         <Toolbar />
-        <div>{children}</div>
+        {children}
```
Comment (apedroferreira) [inline packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx:477]: I guess we don't need a `div` here (unrelated though)

File: packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx [hunk 3]
```diff
@@ -466,9 +466,15 @@ function DashboardLayout(props: DashboardLayoutProps) {
       >
         {drawerContent}
       </Drawer>
-      <Box component="main" sx={{ flexGrow: 1 }}>
+      <Box
+        component="main"
+        sx={{
+          flexGrow: 1,
+          minWidth: { xs: isMobileNavigationOpen ? '100vw' : 'auto', md: 'auto' },
```
Comment (Janpot) [inline packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx:473]: Let's add a comment to the github issue on mui core repository that tracks the issues with the responsive drawer demo.

**Review by Janpot — APPROVED** [review]
Looks good. Just add a comment so that we remember what this hack is for and know when we can revert it.

File: packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx (hunk 3 above)
Comment (apedroferreira) [reply packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx:478]: Raised the issue in https://github.com/mui/material-ui/issues/43244 and added todo comments in https://github.com/mui/mui-toolpad/pull/3905/commits/5658dc02bbb7ddbdae6a513e1ad1fa8813d0521e

**Pull request merged**